container/glWrapper.nix

```diff
@@ -0,0 +1,27 @@
+{ pkgs }:
+
+pkgs.writeTextFile rec {
+  name = "nixGL";
+  text = ''#!${pkgs.runtimeShell}
+
+  vers=$(${pkgs.gnugrep}/bin/grep "Module" /proc/driver/nvidia/version |\
+        ${pkgs.gnused}/bin/sed -E "s/.*Module  ([0-9.]+)  .*/\1/")
+  if [ -z "$vers" ]; then
+    echo "Failed to find your driver version"
```

Comment:
I suggest a more informative error, like "nixGL not built with support for your system driver"

Reply:
that's not what this checks, it checks whether it can parse the version out of /proc, it could fail because the the system doesn't use an nvidia driver, because the content of that file changes formatting, but not because the environment wasn't built with support for that particular version. But "Failed to parse driver version from $file: does $file exist?" would probably be more obvious

Reply:
oh right sorry.

should we add such a check before calling `exec` on a possibly nonexistent script below though?

Reply:
what happens if we exec a non-existent program?

Reply:
nothing bad, just you get `nixGL-<...> not found` while `not compiled for nvidia driver <...>` might be more useful for users

Reply:
so something simple as `exec <prog> "$@" || echo "Environment does not provider a wrapper for driver version $vers && exit 1` ?

Reply:
yeah exactly

Reply:
oh wait it should specifically check for the existence of <prog> not give this error for any nonzero return sorry

Reply:
exec propagates the error code of the called program